sys/log/full/src/log_fcb.c

```diff
@@ -364,48 +383,31 @@ log_fcb_append_mbuf(struct log *log, const struct os_mbuf *om)
 }
 
 static int
-log_fcb_append_mbuf_body(struct log *log, const struct log_entry_hdr *hdr,
-                         const struct os_mbuf *om)
+log_fcb_append_mbuf(struct log *log, struct os_mbuf *om)
 {
-    struct fcb *fcb;
-    struct fcb_entry loc;
-    struct fcb_log *fcb_log;
-    int len;
-    int rc;
-
-    fcb_log = (struct fcb_log *)log->l_arg;
-    fcb = &fcb_log->fl_fcb;
-
-    /* This function expects to be able to write each mbuf without any
-     * buffering.
-     */
-    if (fcb->f_align != 1) {
-        return SYS_ENOTSUP;
-    }
+    struct os_mbuf m;
+    uint16_t mlen;
+    uint16_t hdr_len;
+    struct log_entry_hdr hdr;
 
-    len = sizeof *hdr + os_mbuf_len(om);
-    rc = log_fcb_start_append(log, len, &loc);
-    if (rc != 0) {
-        return rc;
+    mlen = os_mbuf_len(om);
+    if (mlen < LOG_BASE_ENTRY_HDR_SIZE) {
+        return SYS_ENOMEM;
     }
 
-    rc = flash_area_write(loc.fe_area, loc.fe_data_off, hdr, sizeof *hdr);
-    if (rc != 0) {
-        return rc;
-    }
-    loc.fe_data_off += sizeof *hdr;
+    memcpy(&m, om, sizeof m);
 
-    rc = log_fcb_write_mbuf(&loc, om);
-    if (rc != 0) {
-        return rc;
-    }
+    om = os_mbuf_pullup(om, LOG_BASE_ENTRY_HDR_SIZE);
+    
+    hdr_len = log_hdr_len((struct log_entry_hdr *)om->om_data);
+    
+    om = os_mbuf_pullup(om, hdr_len);
+    
+    memcpy(&hdr, om->om_data, hdr_len);
 
-    rc = fcb_append_finish(fcb, &loc);
-    if (rc != 0) {
-        return rc;
-    }
+    os_mbuf_adj(&m, hdr_len);
 
-    return 0;
+    return log_fcb_append_mbuf_body(log, &hdr, &m);
 }
```

Comment:
As with log_cbmem_append_mbuf():

* Should use OS_MBUF_PKTLEN()
* Some comments would help a lot!

Reply:
Not applicable here.